### ISSUE TITLE
template.yml inside folder breaks 'sam local invoke'... still

### ISSUE THREAD

**Issue by red8888** [awslabs/aws-sam-cli#487]
see this: https://github.com/awslabs/aws-sam-cli/issues/140
Same problem with version: version 0.3.0

Uninstalled sam npm, upgraded to latest sam local with pip (python 2)

Windows 10.

```
sam local invoke blah --template mytemps/mytemp.yaml -e myevents/test.json
Unable to import module 'lambda_function': No module named 'lambda_function'
```

Works when yaml template is in the root of the folder. Do I have to use codeUri or something?

YAML
```
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::Serverless-2016-10-31'
Description: An AWS Serverless Specification template describing your function.
Resources:
  blah:
    Type: 'AWS::Serverless::Function'
    Properties:
      Handler: lambda_function.lambda_handler
      Runtime: python3.6
      Description: ''
      MemorySize: 128
      Timeout: 300
      Role: '......'
      Environment:
        Variables:
          sdfsdfs: sdfsdfsdf

```

**Comment by red8888** [comment]
Welp im dumb looks like this _is_ working. i just needed to copy my code to a different folder and use CodeUri to point to it.

**Comment by red8888** [comment]
OK sorry this is still broken.

I didn't realize i was doing this, but if you copy the sam template to the directory of the CodeUri it works- which is... quite odd. 

So for example you have your template in a subfolder named "mytemps". If you copy your code to another subfolder named "mycode" , have CodeUri point to `../mycode` then specify `--template mytemps/mytemp.yaml`  to SAM local it will fail. BUT if you _also_ copy mytemp.yaml to the root of the "mycode" folder it _will_ work.

This probably has to do with how sam local copies files into the container yah? 

Either way this is an exceptable workaround for me for now but it is odd behavior and not obvious whats happening here.

**Comment by jfuss** [comment]
@red8888 I am having a hard time picturing your setup. Could you provide these details for me?

1. Draw out directory structure
```
dir
  code
    template.yaml
    index.js
```
2. What directory are you running the commands from?
3. Can you provide a template snippet? (I know you provided one above but want everything in one place and to make sure it hasn't changed from what you originally posted.)
4. Command that you are running (for the same reason as 3).

Hoping that gives me enough to deeply understand what you setup it.

**Comment by jsheedy** [comment]
I came here with a similar problem and @red8888's suggestion to modify CodeUri helped me solve it.

When running with the template in a subdirectory:

`sam local start-api -p 9090 --host 0.0.0.0 --template project/template.yml`

I got errors about my handler module not being found. When I moved template.yml to the current directory it worked as expected. 

The fix was to update the path of CodeUri
 
from `build/project.zip` to `../build/project.zip`

The structure is like this:
```
root
  project
    template.yml
  build
    project.zip
```
and I run the sam local command from inside the root directory.
